index.js

```diff
@@ -290,5 +290,17 @@ module.exports = {
         ],
       },
     },
+    {
+      files: ['*.d.ts'],
```

Comment:
We added this second rule as a way to override the rule that applies to all `ts` files. We want to set the option `disallowTypeAnnotations: false`, which is `true` by default whenever this rule is enabled. We wanted to override the previous override just for `d.ts` files. We do not want this override to apply to all `ts` and `tsx` files, just to `d.ts`. 😄

Reply:
@angel3cu I guess will you want a beta version released so you can test this before we release?

Reply:
We have tested this in Banana, by updating the package directly in `node_modules` (see screenshots above). If you think a beta version is necessary still, happy to give it a try. 🙂

Reply:
No worries if it has been tested already in some form then happy to go with that!

Reply:
Thanks! Kamil will upload some more screenshots of before and after to show exactly which error we are trying to override and for which files. Essentially, we want to use `type x = import('module').x` in `d.ts` files, because there is no other way of doing this in these files. But in `ts`/`tsx` files we want to enforce `import type x from 'module'` only. 🙂

Reply:
We tested it for the following cases in Banana:

Allows `import()` syntax in `.d.ts` files:
<img width="712" alt="Screenshot 2022-09-23 at 12 56 12" src="https://user-images.githubusercontent.com/56206694/191955120-29ee5271-65cd-4b3a-91cb-8234b8f6655b.png">
Does not allow `import()` in non-`d.ts.` files:
<img width="712" alt="Screenshot 2022-09-23 at 12 56 01" src="https://user-images.githubusercontent.com/56206694/191955108-4669e35f-9dd7-4ed2-a092-7da3331fe2b4.png">